### ISSUE TITLE
Upstreaming this driver

### ISSUE THREAD

**Issue by aleksamagicka** [aleksamagicka/asus_rog_ryujin-hwmon#5]
The driver seems to be in good shape for a v1, and I'd like to send it in for inclusion in the kernel.

@FlorianFreudiger, at the very least I want to credit you in the patch for helping [co-develop and test](https://docs.kernel.org/process/submitting-patches.html#when-to-use-acked-by-cc-and-co-developed-by). If you want to (and I'd encourage it, since you have the device), I'll gladly add you as the author & maintainer of the driver (alongside myself). The full explanation of what a maintainer is can be found [here](https://docs.kernel.org/maintainer/feature-and-driver-maintainers.html).

In short, it means that you're someone who can be contacted about the code and can test changes. That said, hwmon drivers are very low maintenance once they get merged, and in my experience of having a much bigger driver in since August 2021, pretty much nothing happens after. And if nothing else, you get to grab a [nitrokey](https://kernel.nitrokey.com/#:~:text=Nitrokeys%20for%20Linux%20Kernel%20Developers,listed%20in%20the%20MAINTAINERS%20file.) :)

Are you interested? If not, I still want to credit you in the patch.

**Comment by FlorianFreudiger** [comment]
Thank you! While I do very much appreciate it, it's still your code and I'm not at all familiar enough with hwmon or kernel development to be a maintainer.

You can still credit me maybe as a Co-Developer or as a tester. :)
`Florian Freudiger <florian.freudiger@proton.me>`

**Comment by aleksamagicka** [comment]
> You can still credit me maybe as a Co-Developer or as a tester. :)

Deal!

Will prepare a patch tomorrow, hopefully, and post here.

**Comment by aleksamagicka** [comment]
Here is [the commit](https://github.com/aleksamagicka/linux/commit/6e50341cf062f09ab9eec4c6286e90f6109f2638) itself and also the patch in [text form](https://github.com/aleksamagicka/asus_rog_ryujin-hwmon/files/13832110/0001-hwmon-Add-driver-for-ASUS-ROG-RYUJIN-II-360-AIO-cool.txt), please have a look.

[No functional changes](https://github.com/aleksamagicka/asus_rog_ryujin-hwmon/compare/5fe4b32a67f1f281f30b881c851d7fd66667ab20..d92161b5dada6e529bafa5d4dff38acb0975d124) since yesterday, just improved docs and some comments in code.

**Comment by aleksamagicka** [comment]
@FlorianFreudiger, just a heads up, you'll be CC-ed on the email thread of the patch. If maintainers post some comments (which is likely), I'll resolve them, but I think we're generally in good shape code-wise here.

**Comment by aleksamagicka** [comment]
This made it in! Closing.
